### ISSUE TITLE
NCPA RHEL 7 systemd

### ISSUE THREAD

**Issue by metabsd** [NagiosEnterprises/ncpa#349]
Hi,

I don't find any systemd unit file for ncpa. I also thing ncpa have a bug with systemd.

> [root@sltwas0621 ~]# systemctl status ncpa_listener
> ● ncpa_listener.service - LSB: This manages the NCPA Listener service
>    Loaded: loaded (/etc/rc.d/init.d/ncpa_listener; **bad**; vendor preset: disabled)
>    Active: active (running) since Mon 2017-06-05 14:41:12 EDT; 5min ago
>      Docs: man:systemd-sysv-generator(8)
>   Process: 3087 ExecStart=/etc/rc.d/init.d/ncpa_listener start (code=exited, status=0/SUCCESS)
>    CGroup: /system.slice/ncpa_listener.service
>            └─3096 /usr/local/ncpa/ncpa_listener --start

If I compare with another service I should see enabled like cron as an example.

> [root@sltwas0621 ~]# systemctl status crond
> ● crond.service - Command Scheduler
>    Loaded: loaded (/usr/lib/systemd/system/crond.service; **enabled**; vendor preset: enabled)
>    Active: active (running) since Mon 2017-05-29 07:28:35 EDT; 1 weeks 0 days ago
>  Main PID: 18412 (crond)
>    CGroup: /system.slice/crond.service
>            └─18412 /usr/sbin/crond -n

I have the same problem on all our server.

**Comment by jomann09** [comment]
NCPA doesn't currently have a systemd unit file, is it causing some sort of problem on these systems? or just the weird output? It is planned to be added in future versions, maybe even 2.1 if I can get it in there.

**Comment by jomann09** [comment]
Due to the fact we have two services in 2.x I am going to push this off to 3.0 - I had to double check but the "bad" portion in the systemctl output isn't actually a problem, it just refers to the fact that there is no ncpa_listener.service file available. We will take care of this on installation of NCPA 3, which will only be a single service called `ncpa` which will make life easier.

**Comment by SlickHenry** [comment]
For RHEL 7: conditional restarts and reload appear to not be implemented in init.d scripts, they just share the stop/start restart function. It'd be nice to have when the systemd unit files are implemented. Service also autostarts upon install which uses the default config, please don't! 
If you choose to stick with autostart on install, please change the default IP binding to 127.0.0.1 instead of 0.0.0.0 in the ncpa.conf.

**Comment by jomann09** [comment]
@SlickHenry I think that makes sense. I am likely going to bump this up to version 2.1 so we should be able to redo a bit of how all this works.

**Comment by ccztux** [comment]
@jomann09 I agree with @SlickHenry that the services should not be started when NCPA will be installed. I have added a pull request to fix #651 and i could change the autostart behaviour in this context if you like.

**Comment by ericloyd** [comment]
I concur.  Just because I install something doesn't mean I want it started.  Especially not until I've had a chance to fine tune a config, a firewall, a doughnut, or whatever else.

**Comment by jomann09** [comment]
I think we will be changing the setup behavior in NCPA 3, including the default starting and whatnot. NCPA 3 should be coming out relatively soon since the biggest parts are the fact that it's a single service and it'll be python 3.x so we are going to skip some of the features and get it out sooner than later.

**Comment by corynorell** [comment]
It is likely not restricted to RHEL 7, and I believe it is fixed in NCPA 3+.
